tests/test_posterior.py

```diff
@@ -0,0 +1,36 @@
+# Test logic in posterior class. In particular when some distributions have default geometry
+
+import pytest
+import numpy as np
+import cuqi
+
+def test_posterior_should_give_Model_priority_for_geometry():
+    A = cuqi.model.Model(lambda x: x.ravel(), range_geometry=10**2, domain_geometry=cuqi.geometry.Image2D((10, 10)))
+    x = cuqi.distribution.Gaussian(0, 1, geometry=10**2) # Default geometry
+    y = cuqi.distribution.Gaussian(A(x), 1)
+    posterior = cuqi.distribution.Posterior(y.to_likelihood(np.zeros(10**2)), x)
+    assert posterior.geometry == A.domain_geometry
+
+def test_posterior_should_give_prior_priority_for_geometry_if_default_model_geometry():
+    A = cuqi.model.Model(lambda x: x.ravel(), range_geometry=10**2, domain_geometry=(10,10))
+    x = cuqi.distribution.Gaussian(0, 1, geometry=cuqi.geometry.Image2D((10, 10)))
+    y = cuqi.distribution.Gaussian(A(x), 1)
+    posterior = cuqi.distribution.Posterior(y.to_likelihood(np.zeros(10**2)), x)
+    assert posterior.geometry == x.geometry
+
+def test_posterior_should_give_likelihood_priority_for_geometry_if_default_prior_geometry():
+    x = cuqi.distribution.Gaussian(0, 1, geometry=(10,10))
+    y = cuqi.likelihood.UserDefinedLikelihood(logpdf_func=lambda x: x, geometry=cuqi.geometry.Image2D((10, 10)))
+    posterior = cuqi.distribution.Posterior(y, x)
+    assert posterior.geometry == y.geometry
```

Comment:
also here posterior geometry is `_DefaultGeometry2D(100,)`

Reply:
Same here. Fixed it